ocsp/responder/redis/checked_redis_source.go

```diff
@@ -0,0 +1,107 @@
+package redis
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/go-gorp/gorp/v3"
+	"github.com/letsencrypt/boulder/core"
+	"github.com/letsencrypt/boulder/db"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/ocsp/responder"
+	"github.com/letsencrypt/boulder/sa"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/crypto/ocsp"
+)
+
+// dbSelector is a limited subset of the db.WrappedMap interface to allow for
+// easier mocking of mysql operations in tests.
+type dbSelector interface {
+	SelectOne(holder interface{}, query string, args ...interface{}) error
+	WithContext(ctx context.Context) gorp.SqlExecutor
+}
+
+type checkedRedisSource struct {
+	base    *redisSource
+	dbMap   dbSelector
+	counter *prometheus.CounterVec
+	log     blog.Logger
+}
+
+func NewCheckedRedisSource(base *redisSource, dbMap dbSelector, stats prometheus.Registerer, log blog.Logger) *checkedRedisSource {
```

Comment:
Have this ensure that `base != nil`, and return an error if it is. Otherwise the unchecked call to `src.base.Response` on line 69 can panic.

Reply:
I'd still prefer this to return an error, rather than just propagating the nil-ness upwards to be the next Source's problem to handle, but I guess this works.